File: Medium/Python/Rotate-Array.py
1class Solution(object):
2    def rotate(self, nums, k):
3        dic = dict(enumerate(nums)) # create dict of array ex) { (0,1), (1,2) ....}
4        n = len(nums) # max length 
5        for idx in dic:
6            val = dic.get(idx) # get val from dic 
7            if (idx + k) >= n: #if out of bounds use mod for positioning 
8                newIdx = (idx + k) % n  
9                nums[newIdx] = val # update positioning on array in place 
10            elif (idx + k < n): # if in bounds 
11                newIdx = idx + k # add idx + k
12                nums[newIdx] = val # update positioning on array in place 
13
14            

File: Medium/Java/Rotate-Array.java
1import java.util.Map;
2import java.util.HashMap;
3
4class Solution {
5    public void rotate(int[] nums, int k) {
6        Map<Integer, Integer> dict = new HashMap<>(); 
7        int n = nums.length;
8        for(int i = 0; n > i; i++) {
9            dict.put(i, nums[i]);
10        }
11        int newIdx = 0;
12        for(int idx = 0; n > idx; idx++) {
13            int val = dict.get(idx); 
14            if ((idx + k) >= n) {
15                newIdx = (idx + k) % n;
16                nums[newIdx] = val;
17            } 
18            if ((idx + k) < n) {
19                newIdx = idx + k;
20                nums[newIdx] = val;
21            }
22        }
23    }
24}